src/lib/src/select-container.component.ts

```diff
@@ -239,18 +246,45 @@ export class SelectContainerComponent implements AfterViewInit, OnDestroy {
   }
 
   private observeSelectableItems() {
-    // Update the container as well as all selectable items if the list has changed
-    this.$selectableItems.changes.pipe(takeUntil(this.destroy$)).subscribe((items: QueryList<SelectItemDirective>) => {
-      const newList = items.toArray();
-      const removedItems = this._selectedItems.filter(item => !newList.includes(item.value));
+    // Listen for update actions and either select or deselect an item
+    this.updateItems$
+      .pipe(
+        mergeMap(update => {
```

Comment:
Your `addItem` and `removeItem` methods only have to deal with observables because they need to obtain the selected items. I'd suggest refactoring this to use `withLatestFrom(this._selectedItems$)` as you've done in the observable that's composed below this one and pass the latest selected items to `addItem` and `removeItem`, like this:

```ts
withLatestFrom(this._selectedItems$),
map([update, selectedItems] => {
  /* ...*/
  switch (update.type) {
  case UpdateActions.Add:
    if (this.addItem(item.selectedItems)) {
      item.select()));
    }
    break;
  case UpdateActions.Remove:
    if (this.removeItem(item)) {
      item.deselect();
    }
    break;
  /* ... */
```

The implementations of `addItem` and `removeItem` should then not need to deal with observables.